### PR TITLE
storage: Don't offer to mount while formatting in Anaconda mode

### PR DESCRIPTION
**Pull request by mvollmer**
Demo: https://www.youtube.com/watch?v=Vtte11k6aQM

### CODE REVIEW

File: pkg/storaged/block/format-dialog.jsx [hunk 1]
```diff
@@ -271,8 +271,20 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
...
+        { tag: "nomount", Title: create_partition ? _("Create only") : _("Format only") }
+    ];
+
+    const action_variants_for_empty = [
+        { tag: null, Title: create_partition ? _("Create") : _("Format") }
+    ];
+
+    if (client.in_anaconda_mode()) {
+        action_variants = [
+            { tag: "nomount", Title: create_partition ? _("Create") : _("Format") }
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:285]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240103-141207-4bbc74c3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 2]
```diff
@@ -147,24 +147,35 @@ export const MountPoint = ({ fstab_config, forced_options, backing_block, conten
     }
 
     let extra_text = null;
-    if (!is_filesystem_mounted) {
+    if (client.in_anaconda_mode()) {
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:150]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240103-141207-4bbc74c3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 3]
```diff
@@ -147,24 +147,35 @@
...
         if (!old_dir)
-            extra_text = _("The filesystem has no permanent mount point.");
-        else
-            extra_text = _("The filesystem is not mounted.");
-    } else if (backing_block != content_block) {
-        if (!opt_never_auto)
-            extra_text = _("The filesystem will be unlocked and mounted on the next boot. This might require inputting a passphrase.");
+            extra_text = _("The filesystem has no assigned mount point.");
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240103-141207-4bbc74c3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 4]
```diff
@@ -174,7 +185,23 @@
...
             { extra_text }
         </>);
 };
+
+export const mount_point_text = (mount_point, mounted) => {
+    let mp_text;
+    if (mount_point) {
+        mp_text = client.strip_mount_point_prefix(mount_point);
+        if (mp_text == false)
+            return null;
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:197]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240103-141207-4bbc74c3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 5]
```diff
@@ -174,7 +185,23 @@
...
+        if (!mounted && !client.in_anaconda_mode())
+            mp_text = mp_text + " " + _("(not mounted)");
+    } else {
+        if (client.in_anaconda_mode())
+            mp_text = _("(no assigned mount point)");
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240103-141207-4bbc74c3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:150]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240109-123450-67a2b589-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 6]
```diff
@@ -147,24 +147,35 @@ export const MountPoint = ({ fstab_config, forced_options, backing_block, conten
...
-    if (!is_filesystem_mounted) {
+    if (client.in_anaconda_mode()) {
         if (!old_dir)
-            extra_text = _("The filesystem has no permanent mount point.");
-        else
-            extra_text = _("The filesystem is not mounted.");
-    } else if (backing_block != content_block) {
-        if (!opt_never_auto)
-            extra_text = _("The filesystem will be unlocked and mounted on the next boot. This might require inputting a passphrase.");
+            extra_text = _("The filesystem has no assigned mount point.");
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240109-123450-67a2b589-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx [hunk 7]
```diff
@@ -185,9 +195,13 @@ export const mount_point_text = (mount_point, mounted) => {
...
         if (mp_text == false)
             return null;
-        if (!mounted)
+        if (!mounted && !client.in_anaconda_mode())
             mp_text = mp_text + " " + _("(not mounted)");
-    } else
-        mp_text = _("(not mounted)");
+    } else {
+        if (client.in_anaconda_mode())
+            mp_text = _("(no assigned mount point)");
```
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240109-123450-67a2b589-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240110-085306-03cea355-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240110-085306-03cea355-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240110-152114-ce4995d4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240110-152114-ce4995d4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
I'll resolve the pixel test merge conflict once #19800 is in.

**Review by KKoukiou — APPROVED** [review]

File: pkg/storaged/filesystem/utils.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240115-102744-6891849b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240115-102744-6891849b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:152]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240115-125110-9acae421-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/filesystem/utils.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/filesystem/utils.jsx:202]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19807-20240115-125110-9acae421-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Pull request merged**